### PR TITLE
Validations and List Contributors

### PR DESCRIPTION
**Pull request by briankung**
**Problem:**

1. List moderators couldn't select a contributor when editing a list
2. Validations on listmemberships will break the app - not that we really have any

**Solution:** Invert the condition to allow selecting a contributor and build the listmembership before saving the list

**Complications:**

**Feature Changelog:**

- [bug] Allows selecting contributor while editing list
- [bug] Enable validations and fix associated bugs

### CODE REVIEW

**Comment by wsf246** [comment]
So it's intentional they can't add contributors while creating a new list right?

File: app/views/lists/_form.html.erb [hunk 1]
```diff
@@ -75,7 +75,7 @@
           members = if list.persisted? then @list.members else [current_user] end
           non_members = all_users - members
         %>
-        <%= f.select :members, non_members.map(&:username), {include_blank: 'None'}, {class: "form-control", disabled: current_user.can_moderate?(list)}  %>
+        <%= f.select :members, non_members.map(&:username), {include_blank: 'None'}, {class: "form-control", disabled: !current_user.can_moderate?(list)}  %>
```
Comment (wsf246) [inline app/views/lists/_form.html.erb]: This disables the creator from adding contributors on 'New List'

Comment (briankung) [reply app/views/lists/_form.html.erb]: Dammit

**Comment by briankung** [comment]
Should be good to go now 👍

**Pull request merged**